Add Back Button to Final "Add Machine" Screen

ryan-spencer1220 opened the pull request:
Hey Ryan, thanks for the feedback on the issue! Totally understand your reasoning. Feel free to merge or close if you think this isn’t necessary, I just wanted to give it a shot. If you do decide to implement, et me know if you’d like any adjustments!

**Description:**
This PR implements the feature discussed in [#594](https://github.com/pinballmap/pbm-react/issues/594).

**Changes made:**

- Added a back button to the top-left of the final "Add Machine" modal screen.
- Uses FontAwesome6 with platform-specific icons (arrow-left for Android, chevron-left for iOS).
- Pressing the back button returns the user to the machine selection screen.
- Pressing the cancel button returns the user to the location details screen.

Rationale:
Provides a clear, consistent way for users to go back and select a different machine without implying that they are cancelling the entire process. This maintains the existing large Cancel button for its current flow, but repurposing it to send the user back to the location screen.

Linked Issue:
Fixes #594

ryan-spencer1220 committed:
updated modal to include back button and updated Cancel button to return to location details

ryan-spencer1220 committed:
updated modal to include back button and updated Cancel button to return to location details

ryan-spencer1220 committed:
updated modal to include back button and updated Cancel button to return to location details

RyanTG commented:
Thanks! We'll review, but it might be a little while. Our current task is a pretty significant feature addition, so we're trying to focus on that for a bit. Once that work is merged we'll review this PR.

RyanTG commented:
Ah, I forgot it was a class component. That is just a relic of old code. We have been migrating to functional components on an as-needed basis. So yeah, we would want to migrate this one, if you want to do it.

ryan-spencer1220 committed:
converted class component to functional component and implement useSafeAreaInstes()

ryan-spencer1220 committed:
remove duplicate import statement

ryan-spencer1220 commented:
Hey @RyanTG,

I converted that class component to a functional component and implemented the useSafeAreaInsets() hook you mentioned. Let me know if you’d recommend any other changes. Thanks!

bpoore commented:
Hi @ryan-spencer1220 and @RyanTG, apologies I initially missed this PR. Appreciate the work you are putting into this!

This screen in particular is our most complicated screens. It is used in two different places, adding a machine to a location as well as adding machines when submitting a new location. On this branch, you will see the experience to add machines to a location isn't working. In addition to not working, the `MultiSelectRow` component is being re-rendered for every machine when pressing on a machine. In order to optimize for performance, you can see we were extending from the base PureComponent class (which allows only the pressed machine row to be re-rendered). To port over that same optimization here, we will need to be reaching for memoization. Depending on how y'all want to proceed, we could back out the conversion to function components currently or we will need to work to get this working properly for the "suggest location" context.

Thank you for your work here, I really appreciate it!

bpoore commented:
Hi @ryan-spencer1220, thank you for asking! We do have a staging server that can be fired up. To interact with that rather than the prod API, in your .env update EXPO_PUBLIC_API_URL to `https://pbmstaging.com/api/v1`. Be sure to restart your local server if you have one running locally when you make this change. You can coordinate with @RyanTG as he will be the one to turn it on / off.

RyanTG commented:
Yep, just let me know! We keep it off most of the time. But all actually map edits during testing should be done on it.
